plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererAutoTriggerService.kt

```diff
@@ -95,28 +91,9 @@ class CodeWhispererAutoTriggerService : CodeWhispererAutoTriggerHandler, Disposa
 
         val coroutineScope = applicationCoroutineScope()
 
-        return when (triggerType) {
-            is CodeWhispererAutomatedTriggerType.IdleTime -> run {
-                coroutineScope.launch {
-                    // TODO: potential race condition between hasExistingInvocation and entering edt
-                    // but in that case we will just return in performAutomatedTriggerAction
-                    while (!CodeWhispererInvocationStatus.getInstance().hasEnoughDelayToInvokeCodeWhisperer() ||
-                        CodeWhispererInvocationStatus.getInstance().hasExistingInvocation()
-                    ) {
-                        if (!isActive) return@launch
-                        delay(CodeWhispererConstants.IDLE_TIME_CHECK_INTERVAL)
-                    }
-                    runInEdt {
-                        if (CodeWhispererInvocationStatus.getInstance().isPopupActive()) return@runInEdt
-                        performAutomatedTriggerAction(editor, CodeWhispererAutomatedTriggerType.IdleTime(), latencyContext)
-                    }
-                }
-            }
```

Comment:
double check we don't need this anymore?

Reply:
we don't need this even in the existing code, I don't see we ever have CodeWhispererAutomatedTriggerType.IdleTime getting triggered. Regardless of whether it's an existing bug, this is currently how it works and since I'm revamping the auto-trigger part, the new auto-trigger doesn't need to trigger on idle since we are triggering a lot, so removing this trigger type and relevant code.